ansible/library/test_facts.py

```diff
@@ -103,28 +104,28 @@ class ParseTestbedTopoinfo():
     '''
     def __init__(self, testbed_file):
         self.testbed_filename = testbed_file
-        self.testbed_topo = {}
+        self.testbed_topo = defaultdict()
 
     def read_testbed_topo(self):
+        CSV_FIELDS = ('conf-name', 'group-name', 'topo', 'ptf_image_name', 'ptf', 'ptf_ip', 'server', 'vm_base', 'dut', 'comment')
         with open(self.testbed_filename) as f:
-            topo = csv.DictReader(f)
+            topo = csv.DictReader(f, fieldnames=CSV_FIELDS)
+
+            # Validate all field are in the same order and are present
+            header = next(topo)
```

Comment:
what if you read an empty file. Will it throw any exceptions?

Reply:
did you mean file that does not exist? Reading empty files, while providing field names would return empty dict as in this case the fields are not being looked up from the first line of the csv file.

Reply:
Yes, the extra validation would throw for one of the following reasons: as a result of assertion failure, empty file (None values dereferences,) or header line is missing